pkg/multus/multus.go

```diff
@@ -636,6 +639,17 @@ func CmdAdd(args *skel.CmdArgs, exec invoke.Exec, kubeClient *k8s.ClientInfo) (c
 			return nil, cmdPluginErr(k8sArgs, netName, "error adding container to network %q: %v", netName, err)
 		}
 
+		res, err := cni100.NewResultFromResult(tmpResult)
+		if err != nil {
+			logging.Errorf("CmdAdd: failed to read result: %v, but proceed", err)
```

Comment:
I don't understand, if there is an error converting, that function returns
	if err != nil {
		return nil, err
	}

and res will be equal to nil. Below you use res.Interfaces and res could be nil

What am I missing?

Reply:
`cni100.NewResultFromResult(tmpResult)` is just convert result to cni100. Let's imagine that tmpResult has valid CNI result, but failed to convert cni100. In such case, multus need to return tmpResult as valid result because NewResultFromResult() failure is just failed to convert 1.0.0.

Do you guarantee that all CNI version result object can be converted 1.0.0 forever?

Reply:
I understand that.

But let me ask you something, what is the value of "res" if the function cni100.NewResultFromResult(tmpResult) returns an error?

Reply:
perhaps we need another pair of eyes, @dougbtv, I'm claiming that the function cni100.NewResultFromResult will return (nil, err) if there is an error with convert. Therefore, with this "res, err := cni100.NewResultFromResult(tmpResult)", res will be equal to "nil" and then the code below will panic "if res.Interfaces != nil || res.IPs != nil"

func NewResultFromResult(result types.Result) (*Result, error) {
	newResult, err := convert.Convert(result, ImplementedSpecVersion)
	if err != nil {
		return nil, err
	}
	return newResult.(*Result), nil
}